backend/DB.ts

```diff
@@ -36,11 +37,11 @@ export const dbReady = new Promise<void>((resolve, reject) => {
           console.log("Finished database migrations");
           resolve();
         });
-      },
-      (err) => {
-        reject(err);
-      }
-    );
+      })
+      .catch((err) => {
```

Comment:
`.then(successCallback, errCallback)` is different from `.then(successCallback).catch(errCallback)` in that if an error occurs within successCallback it will only be passed to errCallback in the second code example (`.catch()`), not the first. So I changed to `.catch()` so we always catch the error. And then I stopped rejecting the promise because we want all code to wait on the connection, even if it takes a few tries, rather than failing after the first try.